apps/sla/models.py

```diff
@@ -54,7 +54,11 @@ def start(self, instance):
         """
         compute SLA start moment for the given instance
         """
-        return self.get_start(getattr(instance, date) for date in self.dates)
+        return self.get_start(
+            getattr(instance, date)
+            for date in self.dates
+            if getattr(instance, date) is not None
```

Comment:
This fix makes sense and I am at the same time hesitant about it :smile: I agree that the situation when the date is not really set would result in a pretty non-obvious error here. On the other hand the error means that the policy is written wrong - eg. we check Major Incident start date on something not necessarily being a Major Incident. Just skipping the date comparison by `is not None` might hide the bogus policy and result in some wrong/empty SLA deadline being set.

I am thinking about a proper fix. What would you say about the following?

Replace the `getattr` with something like `self.get_date` which would do the `getattr`, then check `is (not) None`, and if empty it would raise
https://github.com/RedHatProductSecurity/osidb/blob/master/apps/sla/exceptions.py#L18
with some descriptive error message that according to the policy a set date was expected but it is in fact empty.

Reply:
I think that it makes sense to have a policy with something like
```yaml
start:
  latest:
    - reported date
    - unembargo date
    - major_incident_start_dt
```
for flaws that can be a MI or not, and that `major_incident_start_dt` should be ignored if it's not a MI, which is what would happen with the current code. If we raise an error then we lose this flexibility by enforcing that this policy only works for MI flaws, but for that we already have the `is major incident` condition.

Reply:
One thing is that MIs have a very specific handling and more strict policy so they should not be consider by a standard policies - currently even the CISA MI has its own policy defined

https://github.com/RedHatProductSecurity/osidb/blob/master/apps/sla/sla.yml#L42

The other thing is that the policies should not be very general but rather as specific as possible so people could very easily connect the SLA context with the resulting deadline only by looking at the SLA file. There should ideally be no logic in the code like what to do when something is empty.